6.5.group_by.py

```diff
@@ -0,0 +1,14 @@
+# group_by 6.5.1
+from collections import defaultdict
+
+
+def group_by(func, lst):
+    d = defaultdict(list)
+
+    for item in lst:
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.